### PR TITLE
KAFKA-2872 Fixed addSink method connecting sink with parent source(s)…

### PR DESCRIPTION
**Pull request by bbejeck**
Starting a KafkaStream was getting an error due to the fact that the TopologyBuilder.addSink method was not connecting the sink with it parent(s) processor/sources.  Just needed to wire up the sink with it parent(s) in TopologyBuilder.addSink .

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java [hunk 1]
```diff
@@ -5,9 +5,9 @@
  * The ASF licenses this file to You under the Apache License, Version 2.0
  * (the "License"); you may not use this file except in compliance with
  * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java]: Is this change intentional?

File: streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java [hunk 2]
```diff
@@ -17,23 +17,18 @@
...
 import org.apache.kafka.test.MockStateStoreSupplier;
 import org.junit.Test;
 
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java]: We would like to avoid wildcard import in the code base.

File: streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java (hunk 1 above)
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java]: No not intentions will fix,

File: streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java (hunk 2 above)
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java]: Will change back and update IntelliJ formatting options

**Comment by bbejeck** [comment]
The test failure (kafka.api.SslConsumerTest.testSimpleConsumption) is unrelated to this patch, as it passes locally and no code was modified that would affect the test.  Does this mean the PR needs to be closed and re-opened as describe in the code contribution guidelines?

**Comment by guozhangwang** [comment]
@bbejeck Thanks, the patch LGTM. The previous failure case is a known transient unit test issue, and we are fixing it right now.

**Comment by guozhangwang** [comment]
Merged to trunk.
